docs/reference/environment-variables.rst

```diff
@@ -0,0 +1,101 @@
+*********************
+Environment variables
+*********************
+
+Applications built on craft-application have several environment variables that
+can configure their behaviour. They and the behaviour they modify are listed
+below.
+
+Variables passed to managed builders
+------------------------------------
+
+Several environment variables from the host environment are passed to the
+managed build environment. While an application may adjust these by adjusting
+the ``environment`` dictionary attached to the ``ProviderService``,
+craft-application will by default forward the ``http_proxy``, ``https_proxy``
+and ``no_proxy`` environment variables from the host.
+
+Supported variables
+-------------------
+
+These variables are explicitly supported for user configuration.
+
+.. _env-var-craft-build-environment:
+
+``CRAFT_BUILD_ENVIRONMENT``
+===========================
+
+If the value is ``host``, allows an environment to tell a craft application to run
+directly on the host rather than in managed mode. This method is roughly equivalent
+to using ``--destructive-mode``, but is designed for configurations where the
+application is already being run in an appropriate container or VM, such as
+`Snapcraft rocks <https://github.com/canonical/snapcraft-rocks/>`_ or
+when controlled by a CI system such as `Launchpad <https://launchpad.net>`_.
+
+**WARNING**: Setting the build environment is only recommended if you are aware of
```

Comment:
```suggestion
**CAUTION**: Setting the build environment is only recommended if you are aware of
```

Missed this one on my first review. I'll let you commit it since I don't want my apparently malfeasant CLA signature to also block this PR from passing checks.